app/maptools/recordingmaptool.cpp

```diff
@@ -285,87 +285,82 @@ void RecordingMapTool::setInitialGeometry( const QgsGeometry &newInitialGeometry
     mPoints.push_back( QgsPoint( *pointIt ) );
   }
 
-  setExistingVertices( extractGeometryVertices( mInitialGeometry ) );
-  setMidPoints( extractMidSegmentVertices( mInitialGeometry ) );
-  setHandles( createHandles( mInitialGeometry ) );
+  createNodesAndHandles();
```

Comment:
We miss a call to build recorded geometry from the initial one here.

```suggestion
  createNodesAndHandles();
  rebuildGeometry();
```